### PR TITLE
Provide a method for narrowing down the type of an iterable to list

### PR DESCRIPTION
**Pull request by Valefant**
Pull Request for the issue: https://github.com/robstoll/atrium/issues/462
If something is missing, let me know :-)
______________________________________
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/iterableAssertions.kt [hunk 1]
```diff
@@ -347,3 +347,29 @@ fun <E, T : Iterable<E>> Expect<T>.hasNext(): Expect<T> =
  */
 fun <E, T : Iterable<E>> Expect<T>.hasNotNext(): Expect<T> =
     _logicAppend { hasNotNext(::identity) }
+
+/**
+ * Turns `Expect<E, T : Iterable<E>>` into `Expect<List<E>`.
+ *
+ * The transformation as such is not reflected in reporting.
+ * Use `feature { f(it::asList) }` if you want to show the transformation in reporting.
```
Comment (robstoll) [inline apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/iterableAssertions.kt:355]:
```suggestion
 * Use `feature { f(it::toList) }` if you want to show the transformation in reporting.
```

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/iterableAssertions.kt [hunk 2]
```diff
@@ -347,3 +347,29 @@ fun <E, T : Iterable<E>> Expect<T>.hasNext(): Expect<T> =
...
+ * @since 0.14.0
+ */
+fun <E, T : Iterable<E>> Expect<T>.asList(): Expect<List<E>> = _logic.changeSubject.unreported { it.toList() }
+
+/**
+ * Expects that the subject of the assertion holds all assertions the given [assertionCreator] creates for
+ * the subject as [List].
+ *
+ * The transformation as such is not reflected in reporting.
+ * Use `feature of({ f(it::asList) }, assertionCreator)` if you want to show the transformation in reporting.
```
Comment (robstoll) [inline apis/fluent-en_GB/atrium-api-fluent-en_GB-common/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/iterableAssertions.kt:368]:
```suggestion
 * Use `feature of({ f(it::toList) }, assertionCreator)` if you want to show the transformation in reporting.
```

**Review by robstoll — CHANGES_REQUESTED** [review]

**Comment by robstoll** [comment]
@Valefant thanks for your first contribution :slightly_smiling_face: :+1: 
There were only two small things to fix so that I committed them myself.

**Comment by Valefant** [comment]
@robstoll You're welcome. Yea sure, I would like to continue :-)

**Pull request merged**